### DISCUSSION TITLE
warning about reading pin when writing pin

### DISCUSSION THREAD

**Discussion by rkb965** [posit-conf-2023/vetiver#31] [Q&A] [answered]
Hi all!

I get this warning:
! Please use full name when reading a pin: "vetiver2023/rebeccas-demoniccaracara-inspection-results", not "rebeccas-demoniccaracara-inspection-results".
Writing to pin 'vetiver2023/rebeccas-demoniccaracara-inspection-results'

when writing a new version of my pin?
Store this new model as a new version of the same pin:

```{r}
v <- vetiver_model(inspection_fit, 
                   model_name = "rebeccas-demoniccaracara-inspection-results",
                   description = "a logistic regression predicting food inspection pass/fail on Chicago restaurants")
board |> vetiver_pin_write(v)
```

This doesn't seem too problematic -- but any idea why I get a read error when I think I'm just writing?

Thanks!!

**Comment by rkb965** [comment, marked as answer]
@isabelizimm answered this with modifying the write code to specify my username:
```
v <- vetiver_model(inspection_fit, 
                   model_name = "vetiver2023/rebeccas-demoniccaracara-inspection-results",
                   description = "a logistic regression predicting food inspection pass/fail on Chicago restaurants")
```

but perhaps the warning would be a tiny bit less confusing if it acknowledged that afaik, I am writing, not reading, a pin at this step

**Reply by isabelizimm** [reply]
Thank you so much for this feedback @rkb965 -- you are correct that it is _writing_ the pin in that step, and the error message is backwards. We welcome contributions if you are interested in [opening an issue or pull request!](https://github.com/rstudio/pins-r/issues/new) 🙌

**Comment by juliasilge** [comment]
I opened this issue to fix this message:
https://github.com/rstudio/pins-r/issues/786

**Reply by rkb965** [reply]
I generally appreciate how incredibly speedy you are @juliasilge, but I was hoping to use modifying the tiny little {cli} message as pr practice, so feel free to ignore that issue...

**Reply by juliasilge** [reply]
When you do the PR, add some text that says it will close that issue! 
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
